core/src/consensus/blake_pow/mod.rs

```diff
@@ -164,7 +165,16 @@ impl ConsensusEngine<CodeChainMachine> for BlakePoW {
         let author = *block.header().author();
         let total_reward = self.block_reward(block.header().number())
             + self.block_fee(Box::new(block.transactions().to_owned().into_iter().map(Into::into)));
-        self.machine.add_balance(block, &author, total_reward)
+        self.machine.add_balance(block, &author, total_reward)?;
+
+        let current_metadata = block.state().metadata()?.unwrap();
+        let current_seq = current_metadata.seq();
+        let block_number = block.header().number();
+        let parent_common_seq = self.machine.metadata_seq(Some(block_number - 1));
+        if parent_common_seq != current_seq {
+            self.machine.insert_common_params(block_number, current_seq, current_metadata.params().clone().unwrap());
```

Comment:
What happened when a reorganization has occurred?
Should we remove the common params when the best chain is changed?

Reply:
I modified `insert_common_params` to remove the changing of reorganized blocks.